chapters/lang/items.tex

```diff
@@ -11,10 +11,12 @@ \section{Functions}
 
 \begin{bnf}
   \gd{fundecl} \eq \term{fun} \ \term{\gp{id}} \ \term{(} \ \gtry{\gp{funparams}} \ \term{)} \ \gp{braceblock} \\
-  \gd{funparams} \eq \term{\gp{id}} \ \gmany{ \ \term{,} \ \term{\gp{id}} \ } \ \gtry{\term{,}}
+  \gd{funparams} \eq \term{\gp{expr}} \ \gmany{ \ \term{,} \ \term{expr} \ \term{\gp{expr}} \ } \ \gtry{\term{,}}
 \end{bnf}
 
-A \emph{function}\index{item!function} is a named block, along with optional set of parameters. Functions are declared with the keyword \lstinline{fun}. Function returns the result of evaluating of contained block. Functions may declare a set of input variables as parameters, through which the caller passes arguments into the function. Input variables behave the same as normal variables defined within function block.
+A \emph{function}\index{item!function} is a named block, along with optional set of parameters. Functions are declared with the keyword \lstinline{fun}. If keyword \emph{return} is not used, the function returns the result of evaluating the last expression from the contained block. Functions may declare a set of input variables as parameters, through which the caller passes arguments into the function. Input variables behave the same as normal variables defined within a function block.
+
+A keyword \lstinline{expr} before argument means that it would be evaluated. If it fails, the function will not be executed and return fail \emph{none} term.
```

Comment:
??

Reply:
We talk about it last time.

Reply:
Yes but
1. the grammar is totally borked, it allows only following syntax:

     > fun f(`expr`, `expr`, ...)

    where `expr` are plain tokens

2. Behaviour description uses term `argument` on definition sites. I believe we agreed to use this on call-site, so this shouled be described along with call expressions